File: backend/src/main/java/nlu/fit/backend/dto/user/response/UserResponse.java
package nlu.fit.backend.dto.user.response;

import lombok.Data;

@Data
public class UserResponse {
    private Long id;
    private String fullName;
    private String gender;
    private String phone;
    private String avatarUrl;
    private String avatarPublicId;
    private String status;
}

File: frontend/lib/dtos/user_profile/user_response.dart
class UserResponse {
  final int id;
  final String fullName;
  final String phone;
  final String gender;
  final String? avatarUrl;
  final String? avatarPublicId;

  UserResponse({
    required this.id,
    required this.fullName,
    required this.phone,
    required this.gender,
    this.avatarUrl,
    this.avatarPublicId,
  });

  factory UserResponse.fromJson(dynamic json) {
    return UserResponse(
      id: json['id'],
      fullName: json['fullName'] ?? '',
      phone: json['phone'] ?? '',
      gender: json['gender'] ?? '',
      avatarUrl: json['avatarUrl'] ?? '',
      avatarPublicId: json['avatarPublicId'] ?? '',
    );
  }
}
